<launch>

	<include
		file="$(find autonomous)/launch/camera.launch"
	/>


	<node
		pkg="autonomous"
		name="diagnostic_node"
		type="diagnostic_node.py"
		respawn="true"
		output="screen"
	/>

	
	<node
		pkg="autonomous"
		name="image_preprocessinggray_node"
		type="image_preprocessinggray_node_visual.py"
		respawn="true"
		output="screen"
	/>

	
	<node
		pkg="autonomous"
		name="image_lanedetection_node"
		type="image_lanedetection_node_visual.py"
		respawn="true"
		output="screen"
	/>

	
	<node
		pkg="autonomous"
		name="image_stopsign_node"
		type="image_stopsign_node_visual.py"
		respawn="true"
	/>

	
	<node
		pkg="autonomous"
		name="esc_control_node"
		type="esc_control_node.py"
		respawn="true"
		output="screen"
	/>

	
	<node
		pkg="autonomous"
		name="car_control_node"
		type="car_control_node.py"
		respawn="true"
	/>




	<node
		pkg="rosserial_python"
		name="serial_node"
		type="serial_node.py"
		args="/dev/ttyACM0"
		respawn="true"
	/>

	
	<node name="steering_pid" pkg="pid" type="controller" >
		<param name="Kp" value="3.0" />
		<param name="Ki" value="1.0" />
		<param name="Kd" value="0.1" />
		<param name="upper_limit" value="30" />
		<param name="lower_limit" value="-30" />
		<param name="topic_from_controller" value="control_steering" />
		<param name="topic_from_plant" value="abw_steering" />
		<param name="setpoint_topic" value="set_steering" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="windup_limit" value="10" />
	</node>


</launch>
